### PR TITLE
Allow multiple arguments to be substituted in one run

### PR DESCRIPTION
**Pull request by kumare3**
# TL;DR
A query like
```sql
     SELECT
        	COUNT(*) as total_count
    	FROM
        	hive.events.{{ .Inputs.table }}
    	WHERE
        	ds = '{{ .Inputs.ds }}' AND hr = {{ .Inputs.hr }} AND min = {{ .Inputs.min }}
```
should be substituted. 

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/320

## Follow-up issue
https://github.com/lyft/flyte/issues/321

### CODE REVIEW

File: go/tasks/pluginmachinery/utils/error_collection.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+type ErrorCollection struct {
```
Comment (EngHabu) [inline go/tasks/pluginmachinery/utils/error_collection.go:8]: Can you use this? https://github.com/lyft/flytestdlib/blob/537f86093d86af270aab300742ee1a56f2905885/config/errors.go#L11

File: go/tasks/pluginmachinery/utils/error_collection.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package utils
```
Comment (wild-endeavor) [inline go/tasks/pluginmachinery/utils/error_collection.go:1]: i think maybe plugins doesn't use this but can we move this to https://github.com/lyft/flytestdlib/blob/master/errors/error.go eventually?

File: go/tasks/pluginmachinery/utils/template.go [hunk 3]
```diff
@@ -51,47 +51,51 @@ func ReplaceTemplateCommandArgs(ctx context.Context, command []string, in io.Inp
...
-		return val, errors.Wrapf(err, "failed to bind a value to inputName [%s]", inputName)
-	}
-	if endIdx >= len(val) {
-		return val[:startIdx] + v, nil
-	}
+	totalMatches := 0
+	var errs ErrorCollection
+	val = inputVarRegex.ReplaceAllStringFunc(val, func(s string) string {
+		totalMatches++
+		matches := inputVarRegex.FindAllStringSubmatch(s, 1)
```
Comment (EngHabu) [inline go/tasks/pluginmachinery/utils/template.go:84]: Do we need to do this?

File: go/tasks/pluginmachinery/utils/error_collection.go (hunk 1 above)
Comment (kumare3) [reply go/tasks/pluginmachinery/utils/error_collection.go:8]: i saw that, its not in the errors package but in config. isn't that weird?

File: go/tasks/pluginmachinery/utils/template.go (hunk 3 above)
Comment (kumare3) [reply go/tasks/pluginmachinery/utils/template.go:84]: yup i will remove

File: go/tasks/pluginmachinery/utils/error_collection.go (hunk 2 above)
Comment (kumare3) [reply go/tasks/pluginmachinery/utils/error_collection.go:1]: yes i agree we should move, its replicated in a few places. I will create an issue

**Pull request merged**